src/metrics/scrapper.py

```diff
@@ -1,32 +1,125 @@
         self._url = url
         self._namespace = namespace
+        self._out_folder = out_folder
         self._metrics = metrics
         # TODO make interval match value in cluster
-        self._template = "irate($metric{namespace=$namespace}[3m])"
+        self._template = 'irate($metric{namespace=$namespace}[3m])'
 
-    def make_queries(self):
+    def query_and_dump_metrics(self):
         for metric in self._metrics:
-            query = self._template.replace("$metric", metric)
-            query = query.replace("$namespace", self._namespace)
-            promql = scrape_utils.create_promql(self._url, query, 1, 60)
-            logger.info(f"Promql: {promql}")
-            response = requests.get(promql)
-            logger.info(f"Response: {response.status_code}")
+            logger.info(f'Querying {metric}')
+            promql = self._create_query(metric)
 
+            result = self._make_query(promql)
+            if result.is_err():
+                logger.warning(f'Error querying {metric}. {result.err_value}')
 
+            response = result.ok_value
+            logger.info(f'Response: {response.status_code}')
+            data = response.json()['data']
 
+            logger.info(f'Dumping {metric} data to .csv')
+            self._dump_data(metric, data)
 
+    def _create_query(self, metric: str) -> str:
+        query = self._template.replace('$metric', metric)
+        query = query.replace('$namespace', self._namespace)
+        promql = scrape_utils.create_promql(self._url, query, 1, 60)
+
+        return promql
+
+    def _make_query(self, promql: str) -> Result[Response, str]:
+        try:
+            response = requests.get(promql, timeout=30)
+        except requests.exceptions.Timeout:
+            return Err(f'Timeout error.')
+
+        if response.ok:
+            return Ok(response)
+        return Err(f'Error in query. Status code {response.status_code}. {response.content}')
+
+    def _dump_data(self, metric: str, data: Dict):
+        df = self._create_dataframe_from_data(data)
+        df = self._sort_dataframe(df)
+
+        result = self._prepare_path(metric)
+        if result.is_err():
+            logger.error(f'{result.err_value}')
+            exit(1)
+
+        df.to_csv(result.ok_value)
+        logger.info(f'{metric} data dumped')
+
+    def _prepare_path(self, metric: str) -> Result[Path, str]:
+        output_file = f'{metric}.csv'
+        output_dir = Path(self._out_folder)
+
+        try:
+            output_dir.mkdir(parents=True, exist_ok=True)
+        except OSError as e:
+            return Err(f'Error creating {output_dir}. {e}')
+
+        return Ok(output_dir / output_file)
+
+    def _create_dataframe_from_data(self, data: Dict) -> pd.DataFrame:
+        final_df = pd.DataFrame()
+        for pod_result_dict in data['result']:
+            column_name = pod_result_dict['metric']['pod'] + '_' + pod_result_dict['metric']['node']
```

Comment:
I'd suggest using f-strings as much as possible as the string formatting standard.
```suggestion
column_name = f"{pod_result_dict['metric']['pod']}_{pod_result_dict['metric']['node']}"
```

Reply:
Oh yes, I tend to use that, but I missed this one. Nice catch.